### PR TITLE
fixes bug 1386781 - fix issues in run_setup_postgres.sh

### PR DESCRIPTION
**Pull request by willkg**
This moves the alembic_config to docker_common.env so it's available in both the
processor and webapp containers.

This adds a urlwait to docker/run_setup_postgres.sh to force it to wait for
postgres db to be ready for connections before trying to setup tables and run
migrations.

This moves the database name to a variable that exists already so it's not
hard-coded.

This removes the code in setupdb_app that prevents logging lines from happening
because that makes understanding what happened and debugging it much much
harder.

### CODE REVIEW

**Comment by willkg** [comment]
This only affects building the dev environment. It doesn't affect server environments.

It fixes all the issues I was having. However, even after running this, the webapp errors out because the Firefox product doesn't exist. Maybe instead of adding fake data, we need to retool this to add a critical mass of real data so that the webapp works.

@peterbe ^^^ Thoughts on what to do about data for the webapp? Maybe we nix the code that generates fake data and add some code to insert real data? Maybe pull it from -prod via existing API endpoints?

**Comment by willkg** [comment]
Things from @g-k's bug comments that should get fixed in this PR, too:

> 1. create oauth creds on https://console.developers.google.com/apis/credentials with authorized js origin http://localhost:8000/ and callback url http://localhost:8000/oauth2/signin/
> 2. copy oauth client id and secret to docker/config/docker_common.env

I can't remember if I had thoughts on how people override variables. If I did, I don't think I wrote it down anywhere. That needs to get written down. One thing that I do remember is that the webapp has an additional env file it looks at because it's using `python-decouple`.

> 8. docker-compose run postgresql psql -h postgresql -U postgres breakpad
> 9. enter aPassword on prompt
> 10. UPDATE auth_user SET is_superuser = true;

This is to create a user account that's got superuser status. `make dockersetup` should handle that, too.

> 12. create a product named Firefox version 1.0 
> 13. create a release for product: Firefox, Version: 1.0, recentish release data

This is what I was getting at in the previous comment. `make dockersetup` should handle this.

Relatedly, `setup` might be a lousy name. The script is supposed to do all the setup for postgres and elasticsearch so that you have a basic functional Socorro running in a docker environment. That includes required data, too.

**Comment by willkg** [comment]
I talked with @peterbe and then made a tracker bug and a bunch of other bugs. I think this is worth landing as is. Making setup_app better and having it create a superuser for the webapp will be covered in another PR. Adding data to the lookup tables will also be covered in another PR.

**Comment by willkg** [comment]
This isn't covered by the tests at all. To test this, you'll need to do something like this:

1. `docker-compose rm -fsv` to stop all running containers, remove the containers, and delete any anonymous volumes they created
2. `make dockerbuild`
3. `make dockersetup`

You shouldn't see any obvious errors in that.

Now you should be able to do:

* `docker-compose up webapp` and connect to http://localhost:8000 with your web browser. You'll get a "product doesn't exist" page. That'll get fixed in another PR.
* `docker-compose up processor` and it'll start up and then do its "nothing to do... waiting 7 seconds" thing.

Assuming all that works, this is good to go.

File: docker/config/docker_common.env [hunk 1]
```diff
@@ -3,6 +3,9 @@
 
 # NOTE: These shouldn't be used on -dev, -stage, or -prod environments.
 
+# alembic
+alembic_config=/app/docker/config/alembic.ini
+
```
Comment (willkg) [inline docker/config/docker_common.env]: This is docker-environment-specific, so it should be in this file.

File: docker/run_setup_postgres.sh [hunk 2]
```diff
@@ -4,23 +4,32 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-# Sets up postgres tables, stored procedures, and such.
+set -e
```
Comment (willkg) [inline docker/run_setup_postgres.sh]: Pulled this from @g-k's branch.

File: docker/run_setup_postgres.sh [hunk 3]
```diff
@@ -4,23 +4,32 @@
...
+# before it does anything giving the user a chance to do a "oh no--don't do
+# that!" kind of thing
+echo "Dropping existing db and creating new db named (${DATABASE})..."
 ./scripts/socorro setupdb \
-                  --database_name=breakpad \
-                  --fakedata \
-                  --fakedata_days=3 \
+                  --database_name="${DATABASE}" \
+                  --dropdb \
                   --createdb
```
Comment (willkg) [inline docker/run_setup_postgres.sh]: After talking with @peterbe, we decided that generating fakedata isn't helpful. What we really want is to populate the lookup tables with appropriate data. That'll get done in another PR.

Comment (willkg) [inline docker/run_setup_postgres.sh]: Also, we decided that we want this script to wipe the db and create a new one. So I added the `--dropdb` flag.

File: socorro/external/postgresql/setupdb_app.py [hunk 4]
```diff
@@ -162,15 +162,6 @@ class SocorroDBApp(App):
...
 
-    @staticmethod
-    def get_application_defaults():
-        """since this app is more of an interactive app than the others, the
-        logging of config information is rather disruptive.  Override the
-        default logging level to one that is less annoying."""
-        return {
-            'logging.stderr_error_logging_level': 40  # only ERROR or worse
-        }
-
```
Comment (willkg) [inline socorro/external/postgresql/setupdb_app.py]: I removed this because it makes it impossible to figure out what `setupdb_app` did. The output it produces it ornery, but it happens once when building a dev environment and it's more important to know what it did. If someone doesn't like it, they can adjust the logging level via the command line.

**Comment by peterbe** [comment]
Before I start digging in to the details; there was an error when I ran `make dockersetup`:
```
▶ make dockersetup
/usr/local/bin/docker-compose run webapp /app/docker/run_setup_postgres.sh
Pulling localstack-s3 (atlassianlabs/localstack:0.4.1)...
0.4.1: Pulling from atlassianlabs/localstack
5bab3fa2f461: Pull complete
Digest: sha256:a6e99e97e6e95c2fc044c30fef0037cc906dcb86f5397d1291144cfde45cf20f
Status: Downloaded newer image for atlassianlabs/localstack:0.4.1
Creating socorro_rabbitmq_1 ...
Creating socorro_elasticsearch_1 ...
Creating socorro_postgresql_1 ...
Creating socorro_localstack-s3_1 ...
Creating socorro_localstack-s3_1
Creating socorro_rabbitmq_1
Creating socorro_elasticsearch_1
Creating socorro_postgresql_1 ... done
Dropping existing db and creating new db named (breakpad)...
2017-08-03 19:00:11,761 INFO - setupdb -  - MainThread - app_name: setupdb
2017-08-03 19:00:11,762 INFO - setupdb -  - MainThread - app_version: 0.2
2017-08-03 19:00:11,762 INFO - setupdb -  - MainThread - current configuration:
2017-08-03 19:00:11,763 INFO - setupdb -  - MainThread - alembic_config: /app/docker/config/alembic.ini
2017-08-03 19:00:11,763 INFO - setupdb -  - MainThread - application: <class 'socorro.external.postgresql.setupdb_app.SocorroDBApp'>
2017-08-03 19:00:11,763 INFO - setupdb -  - MainThread - createdb: True
2017-08-03 19:00:11,764 INFO - setupdb -  - MainThread - database_class: <class 'socorro.external.postgresql.connection_context.ConnectionContext'>
2017-08-03 19:00:11,764 INFO - setupdb -  - MainThread - database_hostname: postgresql
2017-08-03 19:00:11,764 INFO - setupdb -  - MainThread - database_name: breakpad
2017-08-03 19:00:11,765 INFO - setupdb -  - MainThread - database_password: *********
2017-08-03 19:00:11,765 INFO - setupdb -  - MainThread - database_port: 5432
2017-08-03 19:00:11,766 INFO - setupdb -  - MainThread - database_superusername: postgres
2017-08-03 19:00:11,766 INFO - setupdb -  - MainThread - database_superuserpassword: *********
2017-08-03 19:00:11,767 INFO - setupdb -  - MainThread - database_username: postgres
2017-08-03 19:00:11,767 INFO - setupdb -  - MainThread - default_password: *********
2017-08-03 19:00:11,767 INFO - setupdb -  - MainThread - dropdb: True
2017-08-03 19:00:11,767 INFO - setupdb -  - MainThread - fakedata: False
2017-08-03 19:00:11,768 INFO - setupdb -  - MainThread - fakedata_days: 7
2017-08-03 19:00:11,768 INFO - setupdb -  - MainThread - force: False
2017-08-03 19:00:11,769 INFO - setupdb -  - MainThread - logger: <socorro.app.socorro_app.LoggerWrapper object at 0x7f2d8d2b0a50>
2017-08-03 19:00:11,769 INFO - setupdb -  - MainThread - logging.stderr_error_logging_level: 10
2017-08-03 19:00:11,769 INFO - setupdb -  - MainThread - logging.stderr_line_format_string: {asctime} {levelname} - {app_name} - {message}
2017-08-03 19:00:11,769 INFO - setupdb -  - MainThread - logging.syslog_error_logging_level: 40
2017-08-03 19:00:11,770 INFO - setupdb -  - MainThread - logging.syslog_facility_string: user
2017-08-03 19:00:11,770 INFO - setupdb -  - MainThread - logging.syslog_host: localhost
2017-08-03 19:00:11,770 INFO - setupdb -  - MainThread - logging.syslog_line_format_string: {app_name} (pid {process}): {asctime} {levelname} - {threadName} - {message}
2017-08-03 19:00:11,770 INFO - setupdb -  - MainThread - logging.syslog_port: 514
2017-08-03 19:00:11,771 INFO - setupdb -  - MainThread - metrics: <datadog.dogstatsd.base.DogStatsd object at 0x7f2d8bcb78d0>
2017-08-03 19:00:11,771 INFO - setupdb -  - MainThread - metricscfg.statsd_host: statsd
2017-08-03 19:00:11,771 INFO - setupdb -  - MainThread - metricscfg.statsd_port: 8125
2017-08-03 19:00:11,772 INFO - setupdb -  - MainThread - no_roles: False
2017-08-03 19:00:11,772 INFO - setupdb -  - MainThread - no_schema: False
2017-08-03 19:00:11,772 INFO - setupdb -  - MainThread - no_staticdata: False
2017-08-03 19:00:11,772 INFO - setupdb -  - MainThread - on_heroku: False
2017-08-03 19:00:11,773 INFO - setupdb -  - MainThread - read_only_users: breakpad_ro, breakpad, analyst
2017-08-03 19:00:11,773 INFO - setupdb -  - MainThread - read_write_users: postgres, breakpad_rw, monitor
2017-08-03 19:00:11,773 INFO - setupdb -  - MainThread - resource.logging.stderr_error_logging_level: 10
2017-08-03 19:00:11,773 INFO - setupdb -  - MainThread - resource.logging.stderr_line_format_string: {asctime} {levelname} - {app_name} - {message}
2017-08-03 19:00:11,773 INFO - setupdb -  - MainThread - resource.logging.syslog_error_logging_level: 40
2017-08-03 19:00:11,774 INFO - setupdb -  - MainThread - resource.logging.syslog_facility_string: user
2017-08-03 19:00:11,774 INFO - setupdb -  - MainThread - resource.logging.syslog_host: localhost
2017-08-03 19:00:11,774 INFO - setupdb -  - MainThread - resource.logging.syslog_line_format_string: {app_name} (pid {process}): {asctime} {levelname} - {threadName} - {message}
2017-08-03 19:00:11,774 INFO - setupdb -  - MainThread - resource.logging.syslog_port: 514
2017-08-03 19:00:11,775 INFO - setupdb -  - MainThread - resource.postgresql.database_class: <class 'socorro.external.postgresql.connection_context.ConnectionContext'>
2017-08-03 19:00:11,775 INFO - setupdb -  - MainThread - resource.postgresql.database_hostname: postgresql
2017-08-03 19:00:11,775 INFO - setupdb -  - MainThread - resource.postgresql.database_name: breakpad
2017-08-03 19:00:11,775 INFO - setupdb -  - MainThread - resource.postgresql.database_port: 5432
2017-08-03 19:00:11,776 INFO - setupdb -  - MainThread - resource.statsd.statsd_host: statsd
2017-08-03 19:00:11,776 INFO - setupdb -  - MainThread - resource.statsd.statsd_port: 8125
2017-08-03 19:00:11,776 INFO - setupdb -  - MainThread - secrets.postgresql.database_password: *********
2017-08-03 19:00:11,776 INFO - setupdb -  - MainThread - secrets.postgresql.database_username: postgres
2017-08-03 19:00:11,776 INFO - setupdb -  - MainThread - unlogged: False
2017-08-03 19:00:11,777 INFO - setupdb -  - MainThread - Postgres version check section with postgresql://postgres:aPassword@postgresql:5432/postgres
2017-08-03 19:00:11,798 INFO - setupdb -  - MainThread - drop database section with postgresql://postgres:aPassword@postgresql:5432/postgres
drop database breakpad [y/N]:
Traceback (most recent call last):
  File "./scripts/socorro", line 5, in <module>
    SocorroWelcomeApp.run()
  File "/app/socorro/app/socorro_app.py", line 181, in run
    values_source_list=values_source_list
  File "/app/socorro/app/socorro_app.py", line 278, in _do_run
    return_code = fix_exit_code(app_to_run.main())
  File "/app/socorro/app/socorro_app.py", line 530, in main
    return requested_app.main()
  File "/app/socorro/external/postgresql/setupdb_app.py", line 323, in main
    self.config.logger.warn('NOT dropping table')
AttributeError: 'LoggerWrapper' object has no attribute 'warn'
make: [dockersetup] Error 1 (ignored)
/usr/local/bin/docker-compose run webapp /app/docker/run_setup_elasticsearch.sh
Starting socorro_elasticsearch_1 ... done
Starting socorro_localstack-s3_1 ...
Starting socorro_rabbitmq_1 ... done
Setting up Elasticsearch indexes.... done
2017-08-03 19:03:48,441 INFO - setup-supersearch -  - MainThread - app_name: setup-supersearch
2017-08-03 19:03:48,442 INFO - setup-supersearch -  - MainThread - app_version: 1.0
2017-08-03 19:03:48,442 INFO - setup-supersearch -  - MainThread - current configuration:
2017-08-03 19:03:48,443 INFO - setup-supersearch -  - MainThread - application: <class 'setup_supersearch_app.SetupSuperSearchApp'>
2017-08-03 19:03:48,443 INFO - setup-supersearch -  - MainThread - elasticsearch.elasticsearch.elasticsearch_class: <class 'socorro.external.es.connection_context.ConnectionContext'>
2017-08-03 19:03:48,443 INFO - setup-supersearch -  - MainThread - elasticsearch.elasticsearch.elasticsearch_connection_wrapper_class: <class 'socorro.external.es.connection_context.Connection'>
2017-08-03 19:03:48,443 INFO - setup-supersearch -  - MainThread - elasticsearch.elasticsearch.elasticsearch_default_index: socorro
2017-08-03 19:03:48,444 INFO - setup-supersearch -  - MainThread - elasticsearch.elasticsearch.elasticsearch_doctype: crash_reports
2017-08-03 19:03:48,444 INFO - setup-supersearch -  - MainThread - elasticsearch.elasticsearch.elasticsearch_index: socorro%Y%W
2017-08-03 19:03:48,444 INFO - setup-supersearch -  - MainThread - elasticsearch.elasticsearch.elasticsearch_timeout: 30
2017-08-03 19:03:48,444 INFO - setup-supersearch -  - MainThread - elasticsearch.elasticsearch.elasticsearch_timeout_extended: 120
2017-08-03 19:03:48,445 INFO - setup-supersearch -  - MainThread - elasticsearch.elasticsearch.elasticsearch_urls: ['http://elasticsearch:9200']
2017-08-03 19:03:48,445 INFO - setup-supersearch -  - MainThread - elasticsearch.elasticsearch.shards_per_index: 10
2017-08-03 19:03:48,445 INFO - setup-supersearch -  - MainThread - elasticsearch.elasticsearch_class: <class 'socorro.external.es.connection_context.ConnectionContext'>
2017-08-03 19:03:48,445 INFO - setup-supersearch -  - MainThread - elasticsearch.elasticsearch_connection_wrapper_class: <class 'socorro.external.es.connection_context.Connection'>
2017-08-03 19:03:48,445 INFO - setup-supersearch -  - MainThread - elasticsearch.elasticsearch_default_index: socorro
2017-08-03 19:03:48,446 INFO - setup-supersearch -  - MainThread - elasticsearch.elasticsearch_doctype: crash_reports
2017-08-03 19:03:48,446 INFO - setup-supersearch -  - MainThread - elasticsearch.elasticsearch_index: socorro%Y%W
2017-08-03 19:03:48,446 INFO - setup-supersearch -  - MainThread - elasticsearch.elasticsearch_timeout: 30
2017-08-03 19:03:48,446 INFO - setup-supersearch -  - MainThread - elasticsearch.elasticsearch_timeout_extended: 120
2017-08-03 19:03:48,446 INFO - setup-supersearch -  - MainThread - elasticsearch.elasticsearch_urls: ['http://elasticsearch:9200']
2017-08-03 19:03:48,447 INFO - setup-supersearch -  - MainThread - elasticsearch.index_creator_class: <class 'socorro.external.es.index_creator.IndexCreator'>
2017-08-03 19:03:48,447 INFO - setup-supersearch -  - MainThread - logger: <socorro.app.socorro_app.LoggerWrapper object at 0x7ff8412f5390>
2017-08-03 19:03:48,447 INFO - setup-supersearch -  - MainThread - logging.stderr_error_logging_level: 10
2017-08-03 19:03:48,447 INFO - setup-supersearch -  - MainThread - logging.stderr_line_format_string: {asctime} {levelname} - {app_name} - {message}
2017-08-03 19:03:48,448 INFO - setup-supersearch -  - MainThread - logging.syslog_error_logging_level: 40
2017-08-03 19:03:48,448 INFO - setup-supersearch -  - MainThread - logging.syslog_facility_string: user
2017-08-03 19:03:48,448 INFO - setup-supersearch -  - MainThread - logging.syslog_host: localhost
2017-08-03 19:03:48,448 INFO - setup-supersearch -  - MainThread - logging.syslog_line_format_string: {app_name} (pid {process}): {asctime} {levelname} - {threadName} - {message}
2017-08-03 19:03:48,449 INFO - setup-supersearch -  - MainThread - logging.syslog_port: 514
2017-08-03 19:03:48,449 INFO - setup-supersearch -  - MainThread - metrics: <datadog.dogstatsd.base.DogStatsd object at 0x7ff8412c9990>
2017-08-03 19:03:48,449 INFO - setup-supersearch -  - MainThread - metricscfg.statsd_host: statsd
2017-08-03 19:03:48,449 INFO - setup-supersearch -  - MainThread - metricscfg.statsd_port: 8125
2017-08-03 19:03:48,449 INFO - setup-supersearch -  - MainThread - resource.elasticsearch.elasticsearch_class: <class 'socorro.external.es.connection_context.ConnectionContext'>
2017-08-03 19:03:48,450 INFO - setup-supersearch -  - MainThread - resource.elasticsearch.elasticsearch_connection_wrapper_class: <class 'socorro.external.es.connection_context.Connection'>
2017-08-03 19:03:48,450 INFO - setup-supersearch -  - MainThread - resource.elasticsearch.elasticsearch_default_index: socorro
2017-08-03 19:03:48,450 INFO - setup-supersearch -  - MainThread - resource.elasticsearch.elasticsearch_doctype: crash_reports
2017-08-03 19:03:48,450 INFO - setup-supersearch -  - MainThread - resource.elasticsearch.elasticsearch_index: socorro%Y%W
2017-08-03 19:03:48,451 INFO - setup-supersearch -  - MainThread - resource.elasticsearch.elasticsearch_timeout: 30
2017-08-03 19:03:48,451 INFO - setup-supersearch -  - MainThread - resource.elasticsearch.elasticsearch_timeout_extended: 120
2017-08-03 19:03:48,451 INFO - setup-supersearch -  - MainThread - resource.elasticsearch.elasticsearch_urls: ['http://elasticsearch:9200']
2017-08-03 19:03:48,452 INFO - setup-supersearch -  - MainThread - resource.logging.stderr_error_logging_level: 10
2017-08-03 19:03:48,452 INFO - setup-supersearch -  - MainThread - resource.logging.stderr_line_format_string: {asctime} {levelname} - {app_name} - {message}
2017-08-03 19:03:48,452 INFO - setup-supersearch -  - MainThread - resource.logging.syslog_error_logging_level: 40
2017-08-03 19:03:48,452 INFO - setup-supersearch -  - MainThread - resource.logging.syslog_facility_string: user
2017-08-03 19:03:48,453 INFO - setup-supersearch -  - MainThread - resource.logging.syslog_host: localhost
2017-08-03 19:03:48,453 INFO - setup-supersearch -  - MainThread - resource.logging.syslog_line_format_string: {app_name} (pid {process}): {asctime} {levelname} - {threadName} - {message}
2017-08-03 19:03:48,453 INFO - setup-supersearch -  - MainThread - resource.logging.syslog_port: 514
2017-08-03 19:03:48,454 INFO - setup-supersearch -  - MainThread - resource.statsd.statsd_host: statsd
2017-08-03 19:03:48,454 INFO - setup-supersearch -  - MainThread - resource.statsd.statsd_port: 8125
2017-08-03 19:03:48,454 INFO - setup-supersearch -  - MainThread - supersearch_fields_file: /app/scripts/data/supersearch_fields.json
2017-08-03 19:03:48,990 INFO - setup-supersearch -  - MainThread - Created new elasticsearch index: socorro
```

Note, the the setup stopped on that input question about dropping the existing database. I hit Enter to say no.

**Comment by peterbe** [comment]
Another error happens when I try to start the webapp. 
<details>
<summary>Output of `docker-compose up webapp`</summary>
socorro_rabbitmq_1 is up-to-date
socorro_postgresql_1 is up-to-date
socorro_elasticsearch_1 is up-to-date
socorro_localstack-s3_1 is up-to-date
Creating socorro_webapp_1 ...
Creating socorro_webapp_1 ... done
Attaching to socorro_webapp_1
webapp_1         | Copying '/app/webapp-django/crashstats/base/static/img/__utm.gif'
webapp_1         | Copying '/app/webapp-django/crashstats/base/static/img/ajax-loader.gif'
webapp_1         | Copying '/app/webapp-django/crashstats/base/static/img/ajax-loader16x16.gif'
webapp_1         | Copying '/app/webapp-django/crashstats/base/static/img/arrow_collapsed.png'
webapp_1         | Copying '/app/webapp-django/crashstats/base/static/img/arrow_expanded.png'
...
webapp_1         | Post-processed 'js/google-analytics.min.js' as 'js/google-analytics.min.b9911b7bfe92.js'
webapp_1         | Post-processed 'js/topcrashers.min.js' as 'js/topcrashers.min.cafd13d39c2a.js'
webapp_1         | Post-processed 'js/manage-events.min.js' as 'js/manage-events.min.55f5cb3d17bf.js'
webapp_1         | Post-processed 'js/date-filters.min.js' as 'js/date-filters.min.181671246d22.js'
webapp_1         | Post-processed 'js/select2.min.js' as 'js/select2.min.40de687e78fd.js'
webapp_1         |
webapp_1         | 218 static files copied to '/tmp/crashstats-static', 338 post-processed.
webapp_1         | Running webapp. Connect with browser using http://localhost:8000/ .
webapp_1         | Performing system checks...
webapp_1         |
webapp_1         | System check identified no issues (0 silenced).
webapp_1         | Unhandled exception in thread started by <function wrapper at 0x7f18dae876e0>
webapp_1         | Traceback (most recent call last):
webapp_1         |   File "/usr/local/lib/python2.7/site-packages/django/utils/autoreload.py", line 229, in wrapper
webapp_1         |     fn(*args, **kwargs)
webapp_1         |   File "/usr/local/lib/python2.7/site-packages/django/core/management/commands/runserver.py", line 116, in inner_run
webapp_1         |     self.check_migrations()
webapp_1         |   File "/usr/local/lib/python2.7/site-packages/django/core/management/commands/runserver.py", line 168, in check_migrations
webapp_1         |     executor = MigrationExecutor(connections[DEFAULT_DB_ALIAS])
webapp_1         |   File "/usr/local/lib/python2.7/site-packages/django/db/migrations/executor.py", line 19, in __init__
webapp_1         |     self.loader = MigrationLoader(self.connection)
webapp_1         |   File "/usr/local/lib/python2.7/site-packages/django/db/migrations/loader.py", line 47, in __init__
webapp_1         |     self.build_graph()
webapp_1         |   File "/usr/local/lib/python2.7/site-packages/django/db/migrations/loader.py", line 191, in build_graph
webapp_1         |     self.applied_migrations = recorder.applied_migrations()
webapp_1         |   File "/usr/local/lib/python2.7/site-packages/django/db/migrations/recorder.py", line 59, in applied_migrations
webapp_1         |     self.ensure_schema()
webapp_1         |   File "/usr/local/lib/python2.7/site-packages/django/db/migrations/recorder.py", line 49, in ensure_schema
webapp_1         |     if self.Migration._meta.db_table in self.connection.introspection.table_names(self.connection.cursor()):
webapp_1         |   File "/usr/local/lib/python2.7/site-packages/django/db/backends/base/base.py", line 162, in cursor
webapp_1         |     cursor = self.make_debug_cursor(self._cursor())
webapp_1         |   File "/usr/local/lib/python2.7/site-packages/django/db/backends/base/base.py", line 135, in _cursor
webapp_1         |     self.ensure_connection()
webapp_1         |   File "/usr/local/lib/python2.7/site-packages/django/db/backends/base/base.py", line 130, in ensure_connection
webapp_1         |     self.connect()
webapp_1         |   File "/usr/local/lib/python2.7/site-packages/django/db/utils.py", line 98, in __exit__
webapp_1         |     six.reraise(dj_exc_type, dj_exc_value, traceback)
webapp_1         |   File "/usr/local/lib/python2.7/site-packages/django/db/backends/base/base.py", line 130, in ensure_connection
webapp_1         |     self.connect()
webapp_1         |   File "/usr/local/lib/python2.7/site-packages/django/db/backends/base/base.py", line 119, in connect
webapp_1         |     self.connection = self.get_new_connection(conn_params)
webapp_1         |   File "/usr/local/lib/python2.7/site-packages/django/db/backends/postgresql_psycopg2/base.py", line 176, in get_new_connection
webapp_1         |     connection = Database.connect(**conn_params)
webapp_1         |   File "/usr/local/lib/python2.7/site-packages/psycopg2/__init__.py", line 164, in connect
webapp_1         |     conn = _connect(dsn, connection_factory=connection_factory, async=async)
webapp_1         | django.db.utils.OperationalError: could not connect to server: Connection refused
webapp_1         | 	Is the server running on host "localhost" (127.0.0.1) and accepting
webapp_1         | 	TCP/IP connections on port 5432?
webapp_1         | could not connect to server: Cannot assign requested address
webapp_1         | 	Is the server running on host "localhost" (::1) and accepting
webapp_1         | 	TCP/IP connections on port 5432?
webapp_1         |
</details>

And then it just hangs there.

File: docker/run_setup_postgres.sh (hunk 3 above)
Comment (peterbe) [inline docker/run_setup_postgres.sh]: The `--dropdb` causes the the `input('drop database breakpad [y/N]:')` thing to appear. Is that expected?

Comment (willkg) [inline docker/run_setup_postgres.sh]: Yes.

https://github.com/mozilla-services/socorro/pull/3885/files/16adc675ab3b8aa588f936de4673124d13ae803d#diff-ddbf2553890a22d8ff37ba215f1fa887R23

**Comment by willkg** [comment]
^^^ That fixes the error with `.warn()` and also fixes handling for non-zero exit codes which apparently didn't work at all. :(

**Comment by willkg** [comment]
@peterbe ^^^ That fixed the `.warn()` problem plus now if you say "no", it ends the `make dockersetup` thing altogether which shouldn't leave the developer in a weird state.

**Comment by willkg** [comment]
Thank you for reviewing this! Good catch on the `.warn()` and exit code problem!
